### PR TITLE
Fix the MapOptions and ResourceOptions default values.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes
This PR fixes a regression is introduced in https://github.com/mapbox/mapbox-maps-android/pull/269, the default rasterizationMode should be set to IDEOGRAPHS_RASTERIZED_LOCALLY, the default fontFamily should be set to "sans-serif" and constrainMode should be set to HEIGHT_ONLY.

The MapInitOptions default values are listed as below:

```
ResourceOptions:

cachePath: ${context.filesDir.absolutePath}/$DATABASE_NAME, where DATABASE_NAME = "mbx.db"
assetPath: context.filesDir.absolutePath
cacheSize: 50_000_000 (50 MB)

MapOptions:

glyphsRasterizationOptions: IDEOGRAPHS_RASTERIZED_LOCALLY
fontFamily: "sans-serif"
pixelRatio: context.resources.displayMetrics.density
constrainMode(ConstrainMode.HEIGHT_ONLY)
contextMode(ContextMode.UNIQUE)
orientation(NorthOrientation.UPWARDS)
viewportMode(ViewportMode.DEFAULT)
crossSourceCollisions(true)
```

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]

**Review by tobrun — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Review by kiryldz — COMMENTED** [review]
@pengdev add test to not face such regression in future?

**Comment by pengdev** [comment]
@kiryldz @tobrun @Chaoba needs your 👀 again since I added more default values to the MapOptions and ResourceOptions.

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt [hunk 1]
```diff
@@ -49,9 +49,16 @@ data class MapInitOptions constructor(
...
+          .fontFamily(FontUtils.extractValidFont(null))
           .build()
-      ).pixelRatio(context.resources.displayMetrics.density)
+      )
+      .pixelRatio(context.resources.displayMetrics.density)
+      .constrainMode(ConstrainMode.HEIGHT_ONLY)
+      .contextMode(ContextMode.UNIQUE)
+      .orientation(NorthOrientation.UPWARDS)
+      .viewportMode(ViewportMode.DEFAULT)
+      .crossSourceCollisions(true)
```
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:61]: What's that?

**Review by kiryldz — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt (hunk 1 above)
Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:61]: It's taken from the MapAttributeParser.

**Review by Chaoba — APPROVED** [review]

**Pull request merged**